### DISCUSSION TITLE
Help compiling the project

### DISCUSSION THREAD

**Discussion by 4rc4n3s** [doudar/SmartSpin2k#246] [General]
Hello,

I'm trying to compile your project, one more time, with a fresh pull from your repository and I'm still facing the same issue.

I'm using vscode 1.61.2 with platform.io 5.2.2 on Windows 10.

I'm including part of the compiler output below:

Processing debug (platform: espressif32; board: esp32doit-devkit-v1; framework: arduino)
PLATFORM: Espressif 32 (3.3.2) > DOIT ESP32 DEVKIT V1
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 0.0.0+sha.46d5afb
 - tool-esptoolpy 1.30100.210531 (3.1.0)
 - toolchain-xtensa32 2.50200.97 (5.2.0)

Building in debug mode

lib\SS2K\src\sensors\CyclePowerData.cpp:9:23: **fatal error: os/endian.h: No such file or directory**
compilation terminated.
Compiling .pio\build\debug\libf38\SS2K\sensors\SensorData.cpp.o
Compiling .pio\build\debug\libf38\SS2K\sensors\SensorDataFactory.cpp.o
lib\SS2K\src\sensors\FitnessMachineIndoorBikeData.cpp:9:23: fatal error: os/endian.h: No such file or directory
compilation terminated.
lib\SS2K\src\sensors\FlywheelData.cpp:8:23: fatal error: os/endian.h: No such file or directory
compilation terminated.
Compiling .pio\build\debug\lib67f\WiFi\ETH.cpp.o
*** [.pio\build\debug\libf38\SS2K\sensors\FitnessMachineIndoorBikeData.cpp.o] Error 1
*** [.pio\build\debug\libf38\SS2K\sensors\CyclePowerData.cpp.o] Error 1
*** [.pio\build\debug\libf38\SS2K\sensors\FlywheelData.cpp.o] Error 1

--

By modifying a couple of files to specify the include directory in the build flags as well as tweaking some files (library.json and sdkconfig.h), I managed to make your project to compile but I guess it is not the way you did things. Can you please help me?

Thanks again.

Cheers,

4rc4n3s

**Comment by doudar** [comment]
I think what the issue may be is that those files actually need to be added to the gitignore. If you just delete library.json and sdkconfig, does it compile?

If so, I’ll update the gitignore so those aren’t included in the repo.

**Comment by 4rc4n3s** [comment]
Thanks for your answer.

I removed sdkconfig.h from ArduinoCompat/include as you stated.

I also had to modify SS2K/library.json build flags to make it find some headers (os/endian.h). There might be another way though.

"build": {
    "flags": [
        "-std=c++11",
        **"-I../ArduinoCompat/include"**
    ]
  }

Now it seems to be compiling fine.

Thanks.
